lib/netsnmp/varbind.rb

```diff
@@ -81,11 +81,11 @@ def convert_to_asn(typ, value)
                    when :ipaddress then 0
                    when :counter32
                      asn_val = [value].pack("N*")
-                     asn_val = asn_val[1..-1] while asn_val.start_with?("\x00")
+                     asn_val = asn_val[1..-1] while asn_val[0] == "\x00".b && asn_val[1].unpack('B').first != '1'
```

Comment:
consider using:

```ruby
... while asn_val.start_with?("\x00") && !asn_val.start_with?("\x00\x80")
```

Alternatively, you could also store those constants in static variables.

Reply:
@HoneyryderChuck I can't do `!asn_val.start_with?("\x00\x80")` since it doesn't cover the whole range of possible values:

* "\x00\x81"
* "\x00\x82"
* "\x00\x83"
* "\x00\x84"
* etc...